File: python/Recursion&Backtracking/RecursionWithArrays/$01displayArray.py
def displayArray(arr, idx, n):
    if idx == len(arr) :
        return
        
    print(arr[idx])
    displayArray(arr, idx + 1, n) 

def main():
    n = int(input())
    arr = [] 
    for i in range(n) :
        val = int(input())
        arr.append(val)
    
    displayArray(arr, 0, len(arr))

main()

File: DSA_Java/Recursion&Backtracking/Recursion with Arrays/$01displayArray.java
import java.io.*;
import java.util.*;

public class $07displayArr {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in) ;
        int n = sc.nextInt() ;
        
        int[] arr = new int[n] ;
        for(int i = 0; i < arr.length; i++){
            arr[i] = sc.nextInt() ;
        }
        
        displayArr(arr, 0) ;
    }

    public static void displayArr(int[] arr, int idx){
        if(idx == arr.length){
            return ;
        }
        
        System.out.println(arr[idx]) ;
        displayArr(arr, idx + 1) ;
    }

}